### ISSUE TITLE
Merge corpus.ReadyCorpus and corpus.Corpus

### ISSUE THREAD

**Issue by oadams** [persephone-tools/persephone#96]
After recent refactoring, ReadyCorpus only extends Corpus to determine the label set. It would be simpler if `corpus.Corpus.__init__()` just ran `determine_labels()` if the labels argument is None.

Side consideration here is the naming of the new function. The role `corpus.Corpus` takes (now that it is no-longer abstract), is essentially a "ready" corpus after `__init__()` is finished. All the utterance features are preprocessed and in the `feats/` dir, all the transcriptions are preprocessed and segmented correctly in the `labels/` dir. Test and validation sets are now specified in the `valid_prefixes.txt` and `test_prefixes.txt` files. This should be made clear in the documentation and naming of the class. Perhaps `PreprocessedCorpus`, though that's somewhat long. Or it could take the name `ReadyCorpus` again.

**Comment by oadams** [comment]
In doing this, I should also add label validation to `Corpus`. If it isn't passed a `labels` argument, then it should use `determine_labels()`. If it is passed a `labels` argument then it should still run `determine_labels()` and raise an exception if they're different. Adds only slight preprocessing overhead but could save the user a lot of time if a model is halfway through training and then they get a cryptic tensorflow exception.

Also, it might be worth moving `determine_labels()` and other static methods to different modules (`preprocess.labels.py` for `determine_labels()`) because the static methods are cluttering up classes a bit.

**Comment by shuttle1987** [comment]
I was looking at this earlier today and I think `ReadyCorpus` should definitely be removed and the functionality incorporated into `Corpus`

**Comment by oadams** [comment]
Cool, thanks for that input. That confirms my thoughts on the matter.

**Comment by shuttle1987** [comment]
I'm thinking this would be a good thing to tackle in a `Corpus` refactor PR similarly to how we fixed some things in the `Model` refactor (#147)

**Comment by oadams** [comment]
Yeah, that sounds good.
